HW1/P3/P3.py

```diff
@@ -0,0 +1,36 @@
+# Initialize Spark
+from pyspark import SparkContext
+sc = SparkContext()
+sc.setLogLevel("ERROR")
+
+# Helper function - sorts the letters in a word alphabetically
+# Returns sorted letters, original word
+
+def sortWord(val):
+    letters = [v for v in val]
+    letters.sort()
+    letters = ''.join(letters)
+    return letters, val
+
+# Load word list
+wlist = sc.textFile('EOWL_words.txt')
+
+# Use helper function to sort letters alphabetically
+words_anagrams = wlist.map(lambda w: sortWord(w))
+
+# Combine all words that contain the same letters
+anagram_words_grouped = words_anagrams.groupByKey().mapValues(list)
```

Comment:
This is a smart way to do things. I think I did something more silly involving counting each letter. If you just sort the word and then group, you are all set.